source/solvers/gls_navier_stokes.cc

```diff
@@ -1262,6 +1269,267 @@ GLSNavierStokesSolver<dim>::setup_AMG()
   amg_preconditioner->initialize(system_matrix, parameter_ml);
 }
 
+
+template <int dim>
+void
+GLSNavierStokesSolver<dim>::solve_system_gcr_iterative(
```

Comment:
What about moving this into deal.II? I worked on GCR a long time ago (https://github.com/dealii/dealii/pull/12760) but never finished the PR...

Reply:
Hey @peterrum,

This is a good idea. We could definitely add this solver to dealii once it is stable. However, this is a relatively immature implementation. I did some modifications from the classic GCR implementation to reuse more information from the previous correction vector. I made these modifications to learn about how iterative linear solvers work. As such, I don't know the exact properties of the solver other than some benchmarks I made. I still named it GCR since alpha is defined in the same way as in the GCR algorithm if you store 0 previous correction vector. 

Also, I wrote this piece of code to have a test platform that I can modify with additional information I have on the matrix structure in the case of the sharp solver in the future.

I will add some information about the exact algorithm in the theory guide.

Reply:
I finish adding a brief overview of the algorithm to the theory guide. If you have any comments or suggestions, let me know. It is my first time playing with an iterative solver, I may have missed some key elements.